### PR TITLE
updating KT & JS for JAR

### PR DESCRIPTION
**Pull request by EbonyLouis**
### Changes
Based on [this ](https://github.com/TBD54566975/known-customer-credential/issues/33)discussion:
- `KCC-Wallet`: Decodes JAR, Verify the JWT, decode its payload & Process SIOP Request
- `KCC Issuer`: Signs the SIOPv2 Auth Request and sends back in JAR format 
- Updated DID naming 

### Question / Issue

In order to sign the SIOPv2 Auth Request in Kotlin, it needs to be a `JwtClaimsSet` before signing. Is this okay to do or should we update Kotlin so we can sign the Auth Request when it's a `JSON object` ? However, it is allowed in JS. To keep it uniformed this is how its currently being done in this PR: (@KendallWeihe , @decentralgabe , @frankhinek , @tomdaffurn )

```js
  // Sign the SIOPv2 Auth Request
  const siopRequestJwtPayload = {
    sub: issuerBearerDid.uri, // Issuer's Decentralized Identifier string
    iss: issuerBearerDid.uri, // Issuer's Decentralized Identifier string
    iat: Math.floor(Date.now() / 1000), // Issued at
    exp: Math.floor(Date.now() / 1000) + 86400, // Expiration time
    request: siopRequestPayload, // Embed the SIOP request payload 
  };

  try {
    const jwtToken = await Jwt.sign({
      signerDid: issuerBearerDid, 
      payload: siopRequestJwtPayload,
    });
  // Send the SIOPv2 Auth Request in JAR format 
    const queryString = `client_id=${encodeURIComponent(issuerBearerDid.uri)}&request=${encodeURIComponent(jwtToken)}`;
    res.send(queryString);
```

### TO DO:

- update KCC-Wallet-Swift, we need to update `web5-swift` on docs site first since `JWT.verify()` was introduced 2 weeks ago. 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207610264850100

### CODE REVIEW

File: site/docs/tbdex/issuer/kcc/kcc-wallet.mdx [hunk 1]
```diff
@@ -172,7 +172,7 @@ Send an HTTP request to the IDV Service Endpoint:
   ]}
 />
 
-The Issuer will then respond with a URI-encoded [SIOPv2 Authorization Request](https://openid.github.io/SIOPv2/openid-connect-self-issued-v2-wg-draft.html#section-9).
+The Issuer will then respond with a [SIOPv2 Authorization Request](https://openid.github.io/SIOPv2/openid-connect-self-issued-v2-wg-draft.html#section-) in the JWT-Secured Authorization Request (JAR) format.
```
Comment (KendallWeihe) [inline site/docs/tbdex/issuer/kcc/kcc-wallet.mdx:175]: Maybe a good idea to link to the spec 

```suggestion
The Issuer will then respond with a [SIOPv2 Authorization Request](https://openid.github.io/SIOPv2/openid-connect-self-issued-v2-wg-draft.html#section-) in the [JWT-Secured Authorization Request (JAR)](https://www.rfc-editor.org/rfc/rfc9101.html) format.
```

File: site/testsuites/testsuite-javascript/__tests__/tbdex/issuer/kcc/knownCustomerCredentialIssuer.test.js [hunk 2]
```diff
@@ -143,20 +143,31 @@ app.get('/idv/siopv2/initiate', async (req, res) => {
...
-      return `${encodeURIComponent(key)}=${value}`;
-    })
-    .join('&');
+  // Sign the SIOPv2 Auth Request
+  const siopRequestJwtPayload = {
+    sub: issuerBearerDid.uri, // Issuer's Decentralized Identifier string
+    iss: issuerBearerDid.uri, // Issuer's Decentralized Identifier string
+    iat: Math.floor(Date.now() / 1000), // Issued at
+    exp: Math.floor(Date.now() / 1000) + 86400, // Expiration time
+    request: siopRequestPayload, // Embed the SIOPv2 Auth request payload
```
Comment (KendallWeihe) [inline site/testsuites/testsuite-javascript/__tests__/tbdex/issuer/kcc/knownCustomerCredentialIssuer.test.js:152]: @decentralgabe is this right? (the JWT claims, and the `request` in particular)

Comment (EbonyLouis) [reply site/testsuites/testsuite-javascript/__tests__/tbdex/issuer/kcc/knownCustomerCredentialIssuer.test.js:152]: Thanks Kendall I left a question for you guys on this in the description,(read above for a little more info) at the moment I can remove it for JS but not for KT so is it okay for now ? @decentralgabe

File: site/testsuites/testsuite-javascript/__tests__/tbdex/issuer/kcc/knownCustomerCredentialIssuer.test.js [hunk 3]
```diff
@@ -82,15 +82,15 @@ app.get('/idv/siopv2/initiate', async (req, res) => {
...
 // :snippet-start: encodeSiopv2AuthRequestIssuerJs
 app.get('/idv/siopv2/initiate', async (req, res) => {
   // Construct the SIOPv2 Authorization Request
-  const siopRequest = {
-    client_id: issuerDid, // Issuer's Decentralized Identifier
+  const siopRequestPayload = {
+    client_id: issuerBearerDid.uri, // Issuer's Decentralized Identifier string
     scope: 'openid', // Standard OpenID Connect scope
     response_type: 'id_token vp_token', // Expected response formats: ID Token and optionally, Verifiable Presentation Token
     response_uri: 'https://issuer.example.com/siopv2/response', // Endpoint for SIOP response delivery
```
Comment (decentralgabe) [inline site/testsuites/testsuite-javascript/__tests__/tbdex/issuer/kcc/knownCustomerCredentialIssuer.test.js:96]: are we using nonces?

File: site/testsuites/testsuite-javascript/__tests__/tbdex/issuer/kcc/knownCustomerCredentialIssuer.test.js (hunk 2 above)
Comment (decentralgabe) [reply site/testsuites/testsuite-javascript/__tests__/tbdex/issuer/kcc/knownCustomerCredentialIssuer.test.js:152]: this looks right to me

File: site/testsuites/testsuite-javascript/__tests__/tbdex/issuer/kcc/knownCustomerCredentialIssuer.test.js [hunk 4]
```diff
@@ -143,20 +143,31 @@ app.get('/idv/siopv2/initiate', async (req, res) => {
...
-        value = encodeURIComponent(JSON.stringify(value));
-      } else {
-        value = encodeURIComponent(value);
-      }
-      return `${encodeURIComponent(key)}=${value}`;
-    })
-    .join('&');
+  // Sign the SIOPv2 Auth Request
+  const siopRequestJwtPayload = {
+    sub: issuerBearerDid.uri, // Issuer's Decentralized Identifier string
```
Comment (decentralgabe) [inline site/testsuites/testsuite-javascript/__tests__/tbdex/issuer/kcc/knownCustomerCredentialIssuer.test.js:148]: issuer here is the party authenticating correct? not the party we're authenticating with?

Comment (EbonyLouis) [reply site/testsuites/testsuite-javascript/__tests__/tbdex/issuer/kcc/knownCustomerCredentialIssuer.test.js:148]: yes its the party authenticating

File: site/testsuites/testsuite-javascript/__tests__/tbdex/issuer/kcc/knownCustomerCredentialWallet.test.js [hunk 5]
```diff
@@ -9,7 +9,7 @@ import { resolveDid } from '@tbdex/protocol'
 
 const issuerBearerDid = await DidJwk.create();
 const issuerDidUri = issuerBearerDid.uri;
-const customerBearerDid = await DidJwk.create();
+const userBearerDid = await DidJwk.create();
```
Comment (angiejones) [inline site/testsuites/testsuite-javascript/__tests__/tbdex/issuer/kcc/knownCustomerCredentialWallet.test.js:12]: any reason why you changed from customer to user? Customer seems more specific

Comment (EbonyLouis) [reply site/testsuites/testsuite-javascript/__tests__/tbdex/issuer/kcc/knownCustomerCredentialWallet.test.js:12]: I think I overthought it lol when I thought from a Wallet pov I felt like user was better than customer

File: site/testsuites/testsuite-kotlin/src/test/kotlin/docs/tbdex/issuer/kcc/KnownCustomerCredentialIssuerTest.kt [hunk 6]
```diff
@@ -318,15 +318,25 @@ class KnownCustomerCredentialIssuerTest {
...
+                    .issuer(issuerBearerDid.uri) // Issuer's DID
+                    .issueTime(System.currentTimeMillis() / 1000) // Issued time
+                    .expirationTime((System.currentTimeMillis() / 1000) + 86400) // Expiration time 
+                    .misc("request", siopRequest.toString()) // Embed the SIOPv2 Auth request payload 
+                    .build()
 
-                call.respondText(queryString, ContentType.Text.Plain)
+                try {
+                    val jwtToken = Jwt.sign(issuerBearerDid, siopRequestJwtPayload)
+                // Send the SIOPv2 Auth Request in JAR format
```
Comment (angiejones) [inline site/testsuites/testsuite-kotlin/src/test/kotlin/docs/tbdex/issuer/kcc/KnownCustomerCredentialIssuerTest.kt:332]:
```suggestion
                   // Send the SIOPv2 Auth Request in JAR format
```

**Review by angiejones — APPROVED** [review]

**Pull request merged**